landlab/components/resource_redistribution/resource_redistribution.py

```diff
@@ -0,0 +1,738 @@
+        P_sh_regrwth=0.25,
+        Pen=0.05,
+        Pgrz=0.01,
+        P_gr=0.5,
+        sh_max_age=600,
+        sh_seedling_max_age=18,
+        sh_seedling_mor_dis=0.0,
+        sh_mor_dis_low_thresh_age=300,
+        sh_mor_dis_low_slp=0.01,
+        sh_mor_dis_high_slp=0.99,
+        P_sh_fire_mor=0.75,
+        P_gr_fire_mor=1.0,
+        sh_mor_ws_thresh=0.01,
+        gr_mor_ws_thresh=0.08,
+        **kwds,
+    ):
+        """
+        Parameters:
+        ----------
+        grid: RasterModelGrid
+            grid, Landlab's RasterModelGrid object
+        e: float, optional
+            soil erosion at bare soil cell (-)
+        R_low_threshold: float, optional
+            Minimum value of R (-)
+        R_threshold: float, optional
+            Maximum value of R (-)
+        R_dep_threshold: float, optional
+            threshold for deposition during adjustment (-)
+        Rth_gr: float, optional
+            Minimum R for establishment of Grass (-)
+        Rth_sh: float, optional
+            Minimum R for establishment of Shrub (-)
+        P_gr_regrwth: float, optional
+            Probability of grass regrowth in burnt grass patches (-)
+        P_sh_regrwth: float, optional
+            Probability of shrub regrowth in burnt shrub patches (-)
+        Pen: float, optional
+            Probability of shrub establishment (neighborhood) (-)
+        Pgrz: float, optional
+            Probability of shrub establishment
+            (seed dispersal due to herbivores) (-)
+        P_gr: float, optional
+            Probability of grass establishment (-)
+        sh_max_age: int, optional
+            Maximum age of shrubs (yr)
+        sh_seedling_max_age: int, optional
+            Maximum age of shrub seedlings (yr)
+        sh_seedling_mor_dis: float, optional
+            Probability of mortality of shrub seedlings due to disease (-)
+        sh_mor_dis_low_thresh_age: int, optional
+            Age at which shrubs start experiencing disease (yr)
+        sh_mor_dis_low_slp: float, optional
+            Probability of mortality of shrubs due to disease
+            at  sh_mor_dis_low_thresh_age (-)
+        sh_mor_dis_high_slp: float, optional
+            Probability of mortality of shrubs due to disease
+            at sh_max_age (-)
+        P_sh_fire_mor: float, optional
+            Probability of shrub mortality due to fire (-)
+        P_gr_fire_mor: float, optional
+            Probability of grass mortality due to fire (-)
+        sh_mor_ws_thresh: float, optional
+            Background mortality probability of shrubs due
+            to waterstress (P_mor_ws) (-)
+        gr_mor_ws_thresh: float, optional
+            Background mortality probability of grass due
+            to waterstress (-)
+        """
+
+        super(ResourceRedistribution, self).__init__(grid, **kwds)
+
+        name = "vegetation__plant_functional_type"
+        if name not in self.grid.at_cell:
+            print(
```

Comment:
Use *warning.warn* or, if you don't like that, print to *stderr*.

Reply:
done